tests/emcute/main.c

```diff
@@ -0,0 +1,368 @@
+ *
+ * @}
+ */
+
+#include <stdbool.h>
+#include <stdio.h>
+#include <string.h>
+
+#include "net/emcute.h"
+#include "net/mqttsn.h"
+#include "net/ipv6/addr.h"
+#include "od.h"
+#include "shell.h"
+#include "thread.h"
+#include "net/sock/util.h"
+
+/* get to maximum length for client ID ;-)*/
+#define EMCUTE_ID           "emcute test app ......."
+#define EMCUTE_PRIO         (THREAD_PRIORITY_MAIN - 1)
+
+#define NUMOFTOPS           (4U)
+#define SHELL_BUFSIZE       (512U)  /* for sub with long topic */
+
+static char _emcute_stack[THREAD_STACKSIZE_DEFAULT];
+static char _shell_buffer[SHELL_BUFSIZE];
+static uint8_t _pub_buf[EMCUTE_BUFSIZE];
+
+static emcute_topic_t _topics[NUMOFTOPS];
+static emcute_sub_t _subscriptions[NUMOFTOPS];
+static char _topic_names[NUMOFTOPS][EMCUTE_TOPIC_MAXLEN + 1];
+static char _addr_str[IPV6_ADDR_MAX_STR_LEN];
+
+static sock_udp_ep_t _gw = { .family = AF_INET6 };
+
+static int _con(int argc, char **argv);
+static int _discon(int argc, char **argv);
+static int _reg(int argc, char **argv);
+static int _pub(int argc, char **argv);
+static int _sub(int argc, char **argv);
+static int _unsub(int argc, char **argv);
+static int _will(int argc, char **argv);
+static int _info(int argc, char **argv);
+
+static const shell_command_t _shell_commands[] = {
+    { "con", "connect to a MQTT-SN broker", _con },
+    { "discon", "disconnect from current broker", _discon },
+    { "reg", "register to a topic", _reg },
+    { "pub", "publish a number of bytes under a topic", _pub },
+    { "sub", "subscribe to a topic", _sub },
+    { "unsub", "unsubscribe from a topic", _unsub },
+    { "will", "register a last will", _will },
+    { "info", "print client state", _info },
+    { NULL, NULL, NULL },
+};
+
+static void *_emcute_thread(void *arg)
+{
+    (void)arg;
+    emcute_run(MQTTSN_DEFAULT_PORT, EMCUTE_ID);
+    return NULL;    /* should never be reached */
+}
+
+static unsigned _get_qos(const char *str)
+{
+    int qos = atoi(str);
+    switch (qos) {
+        case 1:     return EMCUTE_QOS_1;
+        case 2:     return EMCUTE_QOS_2;
+        default:    return EMCUTE_QOS_0;
+    }
+}
+
+static void _on_pub(const emcute_topic_t *topic, void *data, size_t len)
+{
+    (void)data;
+    printf("### got publication of %u bytes for topic '%s' [%d] ###\n",
+           (unsigned)len, topic->name, (int)topic->id);
+}
+
+static int _con(int argc, char **argv)
+{
+    char *topic = NULL;
+    char *message = NULL;
+    size_t len = 0;
+
+    if (argc < 2) {
+        printf("usage %s <addr> [<will topic> <will msg>]\n",
+               argv[0]);
+        return 1;
+    }
+
+    if (sock_udp_str2ep(&_gw, argv[1]) != 0) {
+        puts("error: unable to parse gateway address");
+        _gw.port = 0;
+        return 1;
+    }
+    if (_gw.port == 0) {
+        _gw.port = MQTTSN_DEFAULT_PORT;
+    }
+    if (argc >= 4) {
+        topic = argv[2];
+        message = argv[3];
+        len = strlen(message);
+    }
+
+
```

Comment:
Vera thinks two consecutive empty lines are too much - I think that should be demoted to a warning (see #13400)

Reply:
why, what's the benefit of 2 empty lines here? I'd rather remove one here.